src/datomish/query/source.cljc

```diff
@@ -72,6 +76,14 @@
             (:fts-view source))]
       [table ((:table-alias source) table)]))
 
+  (source->non-fts-from [source]
```

Comment:
I was thinking, late last night, about special representations for certain data types.  Is it worth trying to generalize a little here?  For example, we might want to add value hashes to certain rows, in order to have fast equality comparisons for large values; or we might want to represent URLs interned (like fulltext, but perhaps stored in some other representation).